File: atcoder/4.cpp
#include <bits/stdc++.h>
#define ll long long
#define f(i, x, n) for(int i = x; i < n; i++)
#define dbg(x) cout << x << endl
#define dbg2(x, y) cout << x << "  " << y << endl
#define dbg3(x, y, z) cout << x << "  " << y << "  " << z << endl
#define mod 1000000007
using namespace std;

int main(){
    ios::sync_with_stdio(0);
    int n;
    cin >> n;
    map<int, int> dict;
    map<int, int> dict2;
    map<int, int>::iterator itr;
    map<int, int>::iterator itr2;
    int arr[n];
    f(i, 0, n){
        cin >> arr[i];
        pair<map<int, int>::iterator, bool> res = dict.insert(pair<int, int>(arr[i], 1));
        if(!res.second){
            dict[arr[i]] += 1;
        }

    }
    for(itr = dict.begin(); itr != dict.end(); ++itr){
        //cout << itr->first << "   " << itr->second << endl;     //////

        int sum = 0;
        if(itr->second > 2){
            sum = ((itr->second -1) * (itr->second - 2)) / 2;
        }
        for(itr2 = dict.begin(); itr2 != dict.end(); ++itr2){
            if(itr != itr2 and itr2->second > 1)
                sum += (itr2->second * (itr2->second -1)) / 2;
        }
        dict2.insert(pair<int, int>(itr->first, sum));
        //dbg(sum);
    }
    f(i, 0, n){
        dbg(dict2[arr[i]]);
    }
    
}

File: atcoder/4.py
n = int(input())
arr = input().split()
arr = [int(i) for i in arr]
dic = {}
for i in arr:
    if i in dic.keys():
        dic[i] += 1
    else:
        dic[i] = 1

dic2 = {}
for i in dic:
    sum = 0
    if dic[i] > 2:
        sum = (dic[i] - 1) * (dic[i] - 2) // 2
    for j in dic:
        if i != j and dic[j] > 1:
            sum += dic[j] * (dic[j] - 1) // 2
    dic2[i] = sum
for i in arr:
    print(dic2[i])